ui/src/components/Equipment/HarvesterMaintenance.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { Row, Col, Button, Card } from 'react-bootstrap';
+
+import { ActionGroup, ActionButton, ActionField } from './ActionGroup';
+import InOutSwitch from '../InOutSwitch/InOutSwitch';
+
+export default function HarvesterMaintenance(props) {
+  const {
+    contents,
+    message,
+    commands_state,
+    currentPlateBarcode,
+    global_state,
+    calibratePin,
+    validateCalibration,
+    sendCommand,
+    commands,
+  } = props;
+
+  function renderActionButton(cmdInfo) {
+    return (
+      <ActionButton
+        label={cmdInfo[1]}
+        cmd={cmdInfo[0]}
+        args={cmdInfo[3]}
+        enabled={commands_state[cmdInfo[0]]}
+        sendCommand={sendCommand}
+        key={cmdInfo[1]}
+      />
+    );
+  }
+
+  function renderActionGroup(grpInfo) {
+    const butGrp = [];
+
+    for (const cmdInfo of grpInfo[1]) {
+      butGrp.push(renderActionButton(cmdInfo));
+    }
+
+    return <ActionGroup name={grpInfo[0]} buttons={butGrp} />;
+  }
+
+  const groups = [];
+  let msg = '';
+
+  if (
+    Object.keys(commands).length > 0 &&
+    commands.cmds !== 'Harvester maintenance controller not defined'
+  ) {
+    for (const cmdgrp of commands.cmds) {
+      groups.push(renderActionGroup(cmdgrp));
+    }
+  } else {
+    return <div />;
+  }
+
+  if (message !== '') {
+    msg = message;
+  }
+
+  let plateBarcode = '';
+  if (global_state.plate_barecode) {
+    plateBarcode = global_state.plate_barecode.toString();
+  }
```

Comment:
I think these things can be simplified to something like the following (and then we have `const`, which I think is preferrable):

```suggestion
  const plateBarcode = global_state.plate_barecode ? global_state.plate_barecode.toString() : "";
```

Also isn't there a formatting convention to always use either single quotes`'` or double quotes `"`?

Reply:
Better like this: thanks
`const plateBarcode = global_state.plate_barecode?.toString();`